backend/migrations/dataall_migrations/migrationmanager.py

```diff
@@ -0,0 +1,67 @@
+import os
+from collections import deque
+from typing import Deque
+from migrations.dataall_migrations.versions.initial import InitMigration
+from migrations.dataall_migrations.base_migration import BaseDataAllMigration
+
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get('LOG_LEVEL', 'INFO'))
+
+
+class MigrationManager:
+    def __init__(self, key='0'):
+        self.current_migration = InitMigration
+        self.previous_migrations: Deque[BaseDataAllMigration] = deque()
+        while self.current_migration.key() != key:
+            self.previous_migrations.append(self.current_migration)
+            self.current_migration = self.current_migration.next_migration()
+        self.executed_upgrades: Deque[BaseDataAllMigration] = deque()
+        self.executed_downgrades: Deque[BaseDataAllMigration] = deque()
+
+    def upgrade(self, target_key=None):
```

Comment:
I see that that we don't invoke upgrade with `target_key`, should we remove it along with all the extra checks/logic around it?

Reply:
I would save it for local testing purposes.

Reply:
Let's assume that we have the migrations `0->1->2->3`, our current revision is `2` (hence ctor key=`2`).
Problems...
* If we call upgrade `target_key=None`
* If we call upgrade `target_key=1` it will execute all the migations to the end
* If we call upgrade 2 times then we will get an exception because `self.current_migration = None`

I suspect there are more combinations of init/up/down that will make this blow up as well.